luaui/barwidgets.lua

```diff
@@ -558,28 +558,21 @@ function widgetHandler:LoadWidget(filename, fromZip, enableLocalsAccess)
 	return widget
 end
 
+local WidgetMeta =
```

Comment:
Why? The tradeoff of using a few extra bytes per widget is probably worth the benefits (can iterate to see what's available, changing the System table doesn't affect interfaces available to existing widgets, metatable existence adds some runtime overhead to all global indexing, metatables look scary to newbie devs)

Reply:
> can iterate to see what's available

You don't have to iterate over widget to know what's there and what isn't. It must be documented to not confuse people.  In addition this kind of `method` is very bad habbit.

> runtime overhead to all global indexing

Upvalue all needed functions within widget and it will be x10 times faster.

> metatables look scary to newbie devs

Metatable is a crutial tool for Lua. If you don't know how it works, first read the manual.